znnl/models/jax_model.py

```diff
@@ -58,34 +83,52 @@ def __init__(
         optimizer : Callable
                 optimizer to use in the training. OpTax is used by default and
                 cross-compatibility is not assured.
-        input_shape : tuple
-                Shape of the NN input.
+        input_shape : Optional[tuple]
+                Shape of the NN input. Required if no pre-built model is passed.
         seed : int, default None
                 Random seed for the RNG. Uses a random int if not specified.
-        ntk_batch_size : int, default 10
+        ntk_batch_size : Optional[int], default 10
                 Batch size to use in the NTK computation.
         trace_axes : Union[int, Sequence[int]]
                 Tracing over axes of the NTK.
                 The default value is trace_axes(-1,), which reduces the NTK to a tensor
                 of rank 2.
                 For a full NTK set trace_axes=().
+        store_on_device : bool, default True
+                Whether to store the NTK on the device or not.
+                This should be set False for large NTKs that do not fit in GPU memory.
+        pre_built_model : Union[None, FlaxPreTrainedModel] (default = None)
+                Pre-built model to use instead of building one from scratch here.
+                So far, this is only implemented for Hugging Face flax models.
         """
+
         self.optimizer = optimizer
         self.input_shape = input_shape
 
         # Initialized in self.init_model
         self.rng = None
 
+        # Input shape is required if no full model is passed.
+        if pre_built_model is None and input_shape is None:
+            raise ValueError(
+                "Input shape must be specified if no pre-built model is passed."
+                "Model is yet to be constructed."
```

Comment:
What do you think about `model has not been constructed` to keep it all in simple English.

Reply:
Fixed in [a87d2e6](https://github.com/zincware/ZnNL/pull/105/commits/a87d2e6bbc614d1986483630d85799c38a57ee47)